scripts/travisnpmpublish.js

```diff
@@ -0,0 +1,28 @@
+const child_process = require('child_process');
+const fs = require('fs');
+
+if (
+  process.env.TRAVIS_BRANCH != 'master' ||
+  (process.env.TRAVIS_PULL_REQUEST && process.env.TRAVIS_PULL_REQUEST != 'false')
+) {
+  console.log('Skipping travis npm publish');
+} else {
+  child_process.exec('git rev-parse --short HEAD', (err, stdout, stderr) => {
+    const hash = stdout.trim();
+    const packageJson = JSON.parse(fs.readFileSync('package.json'));
+    const version = packageJson.version;
+    packageJson.version += `+${hash}`;
+    fs.writeFileSync('package.json', `${JSON.stringify(packageJson, null, 2)}\n`);
+    fs.writeFileSync('.npmrc', '//registry.npmjs.org/:_authToken=' + process.env.NPM_TOKEN);
+    child_process.exec('npm publish', (err) => {
+      if (err) throw err;
+      else
+        child_process.exec(
+          `git tag v${version} && git push -q https://${process.env.GITHUB_TOKEN}@github.com/FredrikNoren/ungit.git v${version}`,
+          (err) => {
+            if (err) throw err;
+          }
+        );
```

Comment:
Unrelated to this specific change, but I feel like braces around this else block would really help readability.